wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Message Details/MessageDetailsDataSource.swift

```diff
@@ -138,14 +153,25 @@ final class MessageDetailsDataSource: NSObject, ZMMessageObserver, ZMUserObserve
         // Detect changes in likes
         if changeInfo.reactionsChanged {
             performChanges {
-                self.reactions = MessageDetailsCellDescription.makeReactionCells(message.sortedLikers)
+                self.reactions = message.usersByReaction.map { reaction, users in
+                    MessageDetailsSectionDescription(
+                        headerText: reaction.unicodeValue,
+                        items: MessageDetailsCellDescription.makeReactionCells(users)
+                    )
+                }.filter {
+                    !$0.items.isEmpty
+                }
```

Comment:
This is not sorted. I suggest to factor this method out to avoid duplication.

Reply:
Done!